// @strapi/design-system/dist/ModalLayout/ModalHeader.js
import o from "react";
import l from "styled-components";
import r from "prop-types";
import { Cross as n } from "@strapi/icons";
import { Flex as i } from "../Flex/Flex.js";
import { Box as p } from "../Box/Box.js";
import { IconButton as m } from "../IconButton/IconButton.js";
import { useModal as s } from "./ModalContext.js";
const c = l(p)`
  border-radius: ${({ theme: e }) => e.borderRadius} ${({ theme: e }) => e.borderRadius} 0 0;
  border-bottom: 1px solid ${({ theme: e }) => e.colors.neutral150};
`, t = ({ children: e, closeLabel: a }) => {
  const d = s();
  return /* @__PURE__ */ o.createElement(c, {
    paddingTop: 4,
    paddingBottom: 4,
    paddingLeft: 5,
    paddingRight: 5,
    background: "neutral100"
  }, /* @__PURE__ */ o.createElement(i, {
    justifyContent: "space-between"
  }, e, /* @__PURE__ */ o.createElement(m, {
    onClick: d,
    "aria-label": a,
    icon: /* @__PURE__ */ o.createElement(n, null)
  })));
};
t.defaultProps = {
  closeLabel: "Close the modal"
};
t.propTypes = {
  children: r.node.isRequired,
  closeLabel: r.string
};
export {
  t as ModalHeader
};
