// @strapi/design-system/dist/Tabs/TabGroup.js
import s, { forwardRef as T, Children as v, useState as I, useImperativeHandle as h } from "react";
import e from "prop-types";
import { TabsContext as x } from "./TabsContext.js";
import { useId as y } from "../helpers/useId.js";
const t = T(({ id: l, initialSelectedTabIndex: d, label: p, onTabChange: b, variant: c, ...i }, m) => {
  const f = y("tabgroup", l), n = v.toArray(i.children).find((a) => a.type.displayName === "Tabs");
  let r = d || 0;
  n && d === void 0 && (r = n.props.children.findIndex((a) => !a.props.disabled));
  const [u, o] = I(r === -1 ? 0 : r);
  return h(m, () => ({
    _handlers: { setSelectedTabIndex: o }
  })), /* @__PURE__ */ s.createElement(x.Provider, {
    value: { id: f, selectedTabIndex: u, selectTabIndex: o, label: p, variant: c, onTabChange: b }
  }, /* @__PURE__ */ s.createElement("div", {
    ...i
  }));
});
t.displayName = "TabGroup";
t.defaultProps = {
  id: void 0,
  initialSelectedTabIndex: void 0,
  onTabChange() {
  },
  variant: void 0
};
t.propTypes = {
  children: e.node.isRequired,
  id: e.string,
  initialSelectedTabIndex: e.number,
  label: e.string.isRequired,
  onTabChange: e.func,
  variant: e.oneOf(["simple"])
};
export {
  t as TabGroup
};
