// @strapi/design-system/dist/LinkButton/LinkButton.cjs
"use strict";Object.defineProperties(exports,{__esModule:{value:!0},[Symbol.toStringTag]:{value:"Module"}});const h=require("react"),$=require("react-router-dom"),S=require("styled-components"),T=require("prop-types"),l=require("../Typography/Typography.cjs"),s=require("../Box/Box.cjs"),r=require("../Button/utils.cjs"),g=require("../Button/constants.cjs"),x=require("../BaseButton/BaseButton.cjs"),u=e=>e&&typeof e=="object"&&"default"in e?e:{default:e},o=u(h),_=u(S),t=u(T),k=_.default(x.BaseButtonWrapper)`
  padding: ${({theme:e,size:a})=>`${a==="S"?e.spaces[2]:"10px"} ${e.spaces[4]}`};
  background: ${({theme:e})=>e.colors.buttonPrimary600};
  border: 1px solid ${({theme:e})=>e.colors.buttonPrimary600};
  border-radius: ${({theme:e})=>e.borderRadius};
  ${s.Box} {
    display: flex;
    align-items: center;
  }
  ${l.Typography} {
    color: ${({theme:e})=>e.colors.buttonNeutral0};
  }
  &[aria-disabled='true'] {
    ${r.getDisabledStyle}
    &:active {
      ${r.getDisabledStyle}
    }
  }
  &:hover {
    ${r.getHoverStyle}
  }
  &:active {
    ${r.getActiveStyle}
  }
  ${r.getVariantStyle}
  /**
    Link specific properties
  */
  display: inline-flex;
  text-decoration: none;
  pointer-events: ${({disabled:e})=>e?"none":void 0};
`,i=o.default.forwardRef(({variant:e,startIcon:a,endIcon:c,disabled:n,children:f,size:p,href:d,to:y,...v},b)=>{const B=d?"_blank":void 0,m=d?"noreferrer noopener":void 0;return o.default.createElement(k,{ref:b,"aria-disabled":n,size:p,variant:e,target:B,rel:m,to:n?void 0:y,href:n?"#":d,...v,as:y&&!n?$.NavLink:"a"},a&&o.default.createElement(s.Box,{"aria-hidden":!0,paddingRight:2},a),p==="S"?o.default.createElement(l.Typography,{variant:"pi",fontWeight:"bold"},f):o.default.createElement(l.Typography,{fontWeight:"bold"},f),c&&o.default.createElement(s.Box,{"aria-hidden":!0,paddingLeft:2},c))});i.displayName="LinkButton";i.defaultProps={disabled:!1,startIcon:void 0,endIcon:void 0,size:"S",variant:"default",onClick:void 0,href:void 0,to:void 0};i.propTypes={children:t.default.node.isRequired,disabled:t.default.bool,endIcon:t.default.element,href(e){if(!e.disabled&&!e.to&&!e.href)return new Error("href must be defined")},onClick:t.default.func,size:t.default.oneOf(g.BUTTON_SIZES),startIcon:t.default.element,to(e){if(!e.disabled&&!e.href&&!e.to)return new Error("to must be defined")},variant:t.default.oneOf(g.VARIANTS)};exports.LinkButton=i;
