// @strapi/design-system/dist/Select/Option.cjs
"use strict";Object.defineProperties(exports,{__esModule:{value:!0},[Symbol.toStringTag]:{value:"Module"}});const s=require("react"),y=require("prop-types"),b=require("styled-components"),n=require("../Box/Box.cjs"),m=require("../Typography/Typography.cjs"),x=require("../Flex/Flex.cjs"),k=require("../BaseCheckbox/assets/checkmark.cjs"),i=e=>e&&typeof e=="object"&&"default"in e?e:{default:e},r=i(s),t=i(y),u=i(b),R=u.default.div`
  border: 1px solid
    ${({theme:e,selected:o,indeterminate:a})=>o||a?e.colors.primary600:e.colors.neutral300};
  border-radius: ${({theme:e})=>e.borderRadius};
  height: 18px;
  width: 18px;
  position: relative;
  z-index: 1;
  overflow: hidden;
  background-color: ${({theme:e,selected:o,indeterminate:a})=>o||a?e.colors.primary600:e.colors.neutral0};

  ${({theme:e,indeterminate:o})=>o&&`&::after {
      content: '';
      display: block;
      position: relative;
      color: white;
      height: 2px;
      width: 10px;
      background-color: ${e.colors.neutral0};
      left: 50%;
      top: 50%;
      transform: translateX(-50%) translateY(-50%);
    }

  `}

  ${({selected:e})=>e&&`  
    &::after {
      content: '';
      background: url(${k}) no-repeat no-repeat center center;
      width: 100%;
      height: 100%;
      position: absolute;
    }

  `}
`,T=u.default(n.Box)`
  width: 100%;
  border: none;
  text-align: left;
  outline-offset: -3px;
  &.is-focused {
    background: ${({theme:e})=>e.colors.primary100};
  }

  &:hover {
    background: ${({theme:e})=>e.colors.primary100};
  }
`,l=({selected:e,indeterminate:o,children:a,value:c,multi:p,isChild:f,startIcon:d,...g})=>{const h=s.useRef(null);return r.default.createElement(T,{as:"li",hasRadius:!0,paddingLeft:f?7:4,paddingRight:4,paddingTop:2,paddingBottom:2,ref:h,role:"option","aria-selected":e,background:"neutral0","data-strapi-value":c,...g},r.default.createElement(x.Flex,null,d&&r.default.createElement(n.Box,{paddingRight:2,"aria-hidden":!0},d),p&&r.default.createElement(n.Box,{paddingRight:2,"aria-hidden":!0},r.default.createElement(R,{selected:e,indeterminate:o})),r.default.createElement(m.Typography,{textColor:e?"primary600":"neutral800",fontWeight:e?"bold":null},a)))};l.defaultProps={isChild:!1,multi:!1,selected:!1,startIcon:void 0,indeterminate:!1};l.propTypes={children:t.default.oneOfType([t.default.string,t.default.number]).isRequired,indeterminate:t.default.bool,isChild:t.default.bool,multi:t.default.bool,selected:t.default.bool,startIcon:t.default.node,value:t.default.oneOfType([t.default.string,t.default.number]).isRequired};l.displayName="Option";exports.Option=l;
