// @strapi/design-system/dist/Select/SelectTags.js
import a from "react";
import e from "prop-types";
import o from "styled-components";
import { Cross as n } from "@strapi/icons";
import { Tag as m } from "../Tag/Tag.js";
import { Flex as i } from "../Flex/Flex.js";
const c = o.div`
  margin-bottom: ${({ theme: r }) => r.spaces[1]};
`, f = o(m)`
  position: relative;
  z-index: 1;
  margin-left: ${({ theme: r }) => r.spaces[1]};
  margin-top: ${({ theme: r }) => r.spaces[1]};
`, l = ({ tags: r, onRemoveTag: s, disabled: p }) => /* @__PURE__ */ a.createElement(c, null, /* @__PURE__ */ a.createElement(i, {
  wrap: "wrap"
}, r.map((t) => /* @__PURE__ */ a.createElement(f, {
  icon: /* @__PURE__ */ a.createElement(n, null),
  disabled: p,
  onClick: () => s(t.value),
  tabIndex: -1,
  key: `tag-${t.value}`
}, t.label))));
l.defaultProps = {
  disabled: !1,
  tags: []
};
l.propTypes = {
  disabled: e.bool,
  onRemoveTag: e.func.isRequired,
  tags: e.arrayOf(
    e.shape({ label: e.string, value: e.oneOfType([e.string, e.number]) })
  )
};
export {
  l as SelectTags
};
