// @strapi/design-system/dist/v2/MainNav/NavSection.js
import e, { Children as m } from "react";
import p from "prop-types";
import { Box as l } from "../../Box/Box.js";
import { Typography as d } from "../../Typography/Typography.js";
import { Stack as t } from "../../Stack/Stack.js";
import { useMainNav as s } from "./MainNavContext.js";
import { VisuallyHidden as c } from "../../VisuallyHidden/VisuallyHidden.js";
import { Divider as u } from "../../Divider/Divider.js";
const g = ({ label: n, children: i, ...o }) => s() ? /* @__PURE__ */ e.createElement(t, {
  spacing: 2
}, /* @__PURE__ */ e.createElement(l, {
  paddingTop: 1,
  paddingBottom: 1,
  background: "neutral0",
  hasRadius: !0,
  as: "span"
}, /* @__PURE__ */ e.createElement(u, null), /* @__PURE__ */ e.createElement(c, null, /* @__PURE__ */ e.createElement("span", null, n))), /* @__PURE__ */ e.createElement(t, {
  as: "ul",
  spacing: 2,
  ...o
}, m.map(i, (r, a) => /* @__PURE__ */ e.createElement("li", {
  key: a
}, r)))) : /* @__PURE__ */ e.createElement(t, {
  spacing: 2
}, /* @__PURE__ */ e.createElement(l, {
  paddingTop: 1,
  paddingBottom: 1,
  background: "neutral0",
  paddingRight: 3,
  paddingLeft: 3,
  hasRadius: !0,
  as: "span"
}, /* @__PURE__ */ e.createElement(d, {
  variant: "sigma",
  textColor: "neutral600"
}, n)), /* @__PURE__ */ e.createElement(t, {
  as: "ul",
  spacing: 2,
  ...o
}, m.map(i, (r, a) => /* @__PURE__ */ e.createElement("li", {
  key: a
}, r))));
g.propTypes = {
  children: p.node.isRequired,
  label: p.string.isRequired
};
export {
  g as NavSection
};
