// @strapi/design-system/dist/NumberInput/NumberInput.js
import t, { useRef as g } from "react";
import r from "prop-types";
import L from "styled-components";
import { CarretDown as v } from "@strapi/icons";
import { NumberParser as T, NumberFormatter as z } from "@internationalized/number";
import { Field as B } from "../Field/Field.js";
import { FieldLabel as U } from "../Field/FieldLabel.js";
import { FieldInput as _ } from "../Field/FieldInput.js";
import { FieldHint as j } from "../Field/FieldHint.js";
import { FieldError as H } from "../Field/FieldError.js";
import "../Field/FieldContext.js";
import "../Field/FieldAction.js";
import { Stack as M } from "../Stack/Stack.js";
import { Icon as N } from "../Icon/Icon.js";
import { sizes as W } from "../themes/sizes.js";
import { useId as X } from "../helpers/useId.js";
import { KeyboardKeys as y } from "../helpers/keyboardKeys.js";
import { getDefaultLocale as Y } from "../helpers/getDefaultLocale.js";
import { useControllableState as G } from "../hooks/useControllableState.js";
const h = L.button`
  display: flex;
  height: 1rem;
  align-items: ${({ reverse: n }) => n ? "flex-end" : "flex-start"};
  transform: translateY(${({ reverse: n }) => n ? "-2px" : "2px"});
  cursor: ${({ disabled: n }) => n ? "not-allowed" : void 0};
  svg {
    display: block;
    height: ${4 / 16}rem;
    transform: ${({ reverse: n }) => n ? "rotateX(180deg)" : void 0};
  }
`, J = "", c = t.forwardRef(
  ({
    size: n,
    startAction: w,
    name: E,
    hint: I,
    error: A,
    label: d,
    labelAction: V,
    locale: x,
    id: D,
    onValueChange: F,
    value: k,
    step: i,
    required: C,
    disabled: s,
    ...P
  }, R) => {
    const O = X("numberinput", D), p = x || Y(), m = g(new T(p, { style: "decimal" })), u = g(new z(p, { maximumFractionDigits: 20 })), [a, S] = G({
      prop(e) {
        const o = String(k);
        return isNaN(o) || o !== e && e !== "" ? e : o;
      },
      defaultProp: J,
      onChange(e) {
        const o = m.current.parse(e);
        F(isNaN(o) ? void 0 : o);
      }
    }), l = (e) => {
      S(String(e));
    }, $ = ({ target: { value: e } }) => {
      m.current.isValidPartialNumber(e) && l(e);
    }, f = () => {
      if (!a) {
        l(i);
        return;
      }
      const e = m.current.parse(a), o = isNaN(e) ? i : e + i;
      l(u.current.format(o));
    }, b = () => {
      if (!a) {
        l(-i);
        return;
      }
      const e = m.current.parse(a), o = isNaN(e) ? -i : e - i;
      l(u.current.format(o));
    }, q = (e) => {
      if (!s)
        switch (e.key) {
          case y.DOWN: {
            e.preventDefault(), b();
            break;
          }
          case y.UP: {
            e.preventDefault(), f();
            break;
          }
        }
    }, K = () => {
      if (a) {
        const e = m.current.parse(a), o = isNaN(e) ? "" : u.current.format(e);
        l(o);
      }
    };
    return /* @__PURE__ */ t.createElement(B, {
      name: E,
      hint: I,
      error: A,
      id: O,
      required: C
    }, /* @__PURE__ */ t.createElement(M, {
      spacing: 1
    }, d && /* @__PURE__ */ t.createElement(U, {
      action: V
    }, d), /* @__PURE__ */ t.createElement(_, {
      ref: R,
      startAction: w,
      disabled: s,
      type: "text",
      inputmode: "decimal",
      onChange: $,
      onKeyDown: q,
      onBlur: K,
      value: a,
      size: n,
      endAction: /* @__PURE__ */ t.createElement(t.Fragment, null, /* @__PURE__ */ t.createElement(h, {
        disabled: s,
        "aria-hidden": !0,
        reverse: !0,
        onClick: f,
        tabIndex: -1,
        type: "button",
        "data-testid": "ArrowUp"
      }, /* @__PURE__ */ t.createElement(N, {
        as: v,
        color: "neutral500"
      })), /* @__PURE__ */ t.createElement(h, {
        disabled: s,
        "aria-hidden": !0,
        onClick: b,
        tabIndex: -1,
        type: "button",
        "data-testid": "ArrowDown"
      }, /* @__PURE__ */ t.createElement(N, {
        as: v,
        color: "neutral500"
      }))),
      ...P
    }), /* @__PURE__ */ t.createElement(j, null), /* @__PURE__ */ t.createElement(H, null)));
  }
);
c.displayName = "NumberInput";
c.defaultProps = {
  "aria-label": void 0,
  disabled: !1,
  error: void 0,
  hint: void 0,
  id: void 0,
  label: void 0,
  labelAction: void 0,
  locale: void 0,
  required: !1,
  size: "M",
  startAction: void 0,
  step: 1,
  value: void 0
};
c.propTypes = {
  "aria-label": r.string,
  disabled: r.bool,
  error: r.string,
  hint: r.oneOfType([r.string, r.node, r.arrayOf(r.node)]),
  id: r.string,
  label: r.string,
  labelAction: r.element,
  locale: r.string,
  name: r.string.isRequired,
  onValueChange: r.func.isRequired,
  required: r.bool,
  size: r.oneOf(Object.keys(W.input)),
  startAction: r.element,
  step: r.number,
  value: r.number
};
export {
  c as NumberInput
};
