// @strapi/design-system/dist/ToggleInput/ToggleInput.js
import r from "react";
import p from "styled-components";
import e from "prop-types";
import { sizes as x } from "../themes/sizes.js";
import { useId as y } from "../helpers/useId.js";
import { Field as T } from "../Field/Field.js";
import { FieldLabel as v } from "../Field/FieldLabel.js";
import "../Field/FieldInput.js";
import { FieldHint as F } from "../Field/FieldHint.js";
import { FieldError as h } from "../Field/FieldError.js";
import "../Field/FieldContext.js";
import "../Field/FieldAction.js";
import { Stack as k } from "../Stack/Stack.js";
import { Flex as C } from "../Flex/Flex.js";
import { TextButton as I } from "../TextButton/TextButton.js";
import { ToggleCheckbox as O } from "../ToggleCheckbox/ToggleCheckbox.js";
const z = p(T)`
  max-width: 320px;
`, q = p(I)`
  align-self: flex-end;
  margin-left: auto;
`, o = ({
  disabled: t,
  size: d,
  error: s,
  hint: c,
  label: i,
  name: f,
  labelAction: g,
  required: u,
  id: b,
  onClear: l,
  clearLabel: n,
  checked: m,
  ...E
}) => {
  const a = y("toggleinput", b);
  return /* @__PURE__ */ r.createElement(z, {
    name: f,
    hint: c,
    error: s,
    id: a,
    required: u
  }, /* @__PURE__ */ r.createElement(k, {
    spacing: 1
  }, /* @__PURE__ */ r.createElement(C, null, /* @__PURE__ */ r.createElement(v, {
    action: g
  }, i), n && l && m !== null && !t && /* @__PURE__ */ r.createElement(q, {
    onClick: l
  }, n)), /* @__PURE__ */ r.createElement(O, {
    id: a,
    size: d,
    checked: m,
    disabled: t,
    ...E
  }, i), /* @__PURE__ */ r.createElement(F, null), /* @__PURE__ */ r.createElement(h, null)));
};
o.displayName = "ToggleInput";
o.defaultProps = {
  checked: !1,
  clearLabel: void 0,
  disabled: !1,
  error: void 0,
  hint: void 0,
  id: void 0,
  label: "",
  labelAction: void 0,
  name: "",
  onClear: void 0,
  required: !1,
  size: "M"
};
o.propTypes = {
  checked: e.bool,
  clearLabel: e.string,
  disabled: e.bool,
  error: e.string,
  hint: e.oneOfType([e.string, e.node, e.arrayOf(e.node)]),
  id: e.string,
  label: e.string,
  labelAction: e.node,
  name: e.string,
  onClear: e.func,
  required: e.bool,
  size: e.oneOf(Object.keys(x.input))
};
export {
  o as ToggleInput
};
