// @strapi/design-system/dist/SubNav/SubNavSection.js
import e, { useState as f, Children as b } from "react";
import t from "prop-types";
import h from "styled-components";
import { Box as l } from "../Box/Box.js";
import { Badge as v } from "../Badge/Badge.js";
import { SubNavSectionLabel as E } from "./SubNavSectionLabel.js";
import { useId as S } from "../helpers/useId.js";
import { Stack as k } from "../Stack/Stack.js";
const C = h(l)`
  svg {
    height: ${4 / 16}rem;
    path {
      fill: ${({ theme: r }) => r.colors.neutral500};
    }
  }
`, p = ({ collapsable: r, label: s, badgeLabel: o, children: d, id: m }) => {
  const [n, c] = f(!0), a = S("subnav-list", m), u = () => {
    c((i) => !i);
  };
  return /* @__PURE__ */ e.createElement(k, {
    spacing: 1
  }, /* @__PURE__ */ e.createElement(C, {
    paddingLeft: 6,
    paddingTop: 1,
    paddingBottom: 1,
    paddingRight: 4
  }, /* @__PURE__ */ e.createElement(l, {
    position: "relative",
    paddingRight: o ? 6 : 0
  }, /* @__PURE__ */ e.createElement(E, {
    onClick: u,
    ariaExpanded: n,
    ariaControls: a,
    collapsable: r,
    label: s
  }), o && /* @__PURE__ */ e.createElement(v, {
    backgroundColor: "neutral150",
    textColor: "neutral600",
    position: "absolute",
    right: 0,
    top: "50%",
    transform: "translateY(-50%)"
  }, o))), (!r || n) && /* @__PURE__ */ e.createElement("ol", {
    id: a
  }, b.map(d, (i, g) => /* @__PURE__ */ e.createElement("li", {
    key: g
  }, i))));
};
p.defaultProps = {
  badgeLabel: null,
  children: void 0,
  collapsable: !1,
  id: void 0
};
p.propTypes = {
  badgeLabel: t.string,
  children: t.node,
  collapsable: t.bool,
  id: t.string,
  label: t.string.isRequired
};
export {
  p as SubNavSection
};
